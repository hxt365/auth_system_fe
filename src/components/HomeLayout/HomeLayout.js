// @flow

import React from 'react';
import { Layout } from 'antd';
import Navigation from 'components/Navigation';
import './HomeLayout.scss';
import Logo from 'assets/pictures/logo.png';

type PropsType = {
  children: any,
};

function HomeLayout(props: PropsType) {
  const { Header, Footer, Content } = Layout;
  const { children } = props;

  return (
    <Layout className="homelayout">
      <Header style={{ position: 'fixed', zIndex: 1, width: '100%' }}>
        <div className="logo">
          <img src={Logo} alt="Teko's logo" />
        </div>
        <Navigation />
      </Header>
      <Content className="content">
        <div className="wrapper">{children}</div>
      </Content>
      <Footer style={{ textAlign: 'center' }}>HXT365 @Teko2019</Footer>
    </Layout>
  );
}

export default HomeLayout;
