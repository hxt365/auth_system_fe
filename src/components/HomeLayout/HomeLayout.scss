.homelayout {
  height: 100vh;

  .content {
    padding: 0 50px;
    margin-top: 6.4rem;
    height: 100%;

    .wrapper {
      background: #fff;
      padding: 24;
      height: 100%;
    }
  }

  .logo {
    display: inline-block;

    height: 100%;
    width: 7rem;

    img {
      height: 100%;
      width: 100%;
    }
  }
}
