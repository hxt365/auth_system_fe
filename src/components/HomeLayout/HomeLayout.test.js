import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import HomeLayout from './HomeLayout';

describe('test HomeLayout', () => {
  it('test snapshot', () => {
    const wrapper = shallow(<HomeLayout />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
