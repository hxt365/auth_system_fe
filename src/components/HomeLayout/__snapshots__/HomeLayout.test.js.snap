// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test HomeLayout test snapshot 1`] = `
<Adapter
  className="homelayout"
>
  <Adapter
    style={
      Object {
        "position": "fixed",
        "width": "100%",
        "zIndex": 1,
      }
    }
  >
    <div
      className="logo"
    >
      <img
        alt="Teko's logo"
        src="logo.png"
      />
    </div>
    <Navigation />
  </Adapter>
  <Adapter
    className="content"
  >
    <div
      className="wrapper"
    />
  </Adapter>
  <Adapter
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    HXT365 @Teko2019
  </Adapter>
</Adapter>
`;
